packages/wmr/src/lib/transform-prefresh-registrations.js

```diff
@@ -0,0 +1,452 @@
+function hash(str) {
+	let hash = 5381,
+		i = str.length;
+	while (i) {
+		hash = (hash * 33) ^ str.charCodeAt(--i);
+	}
+
+	return (hash >>> 0).toString(36);
+}
+
+export default function transformPrefreshRegistrations({ types: t, template }, options = {}) {
+	const refreshReg = t.identifier('$RefreshReg$');
+
+	const registrationsByProgramPath = new Map();
+	const seenForRegistration = new WeakSet();
+	const seenForOutro = new WeakSet();
+
+	const isComponentishName = name => typeof name === 'string' && name[0] >= 'A' && name[0] <= 'Z';
+	let i = 0;
+	function createRegistration(programPath, persistentID) {
+		const handle = t.identifier(`_c${i++}`);
```

Comment:
This is a workaround, in an ideal world we would create a `Scope` class allowing us to generate unique identifiers that aren't just scoped to this particular transpilation plugin, i.e. If we add this twice by accident we'll have conflicting variable names.